implicit-int-float-conversion warnings in test-hexfloat.cc

With:

```
$ clang -v
clang version 10.0.0 (https://github.com/llvm/llvm-project/ 64a362e7216a43e3ad44e50a89265e72aeb14294)
Target: x86_64-unknown-linux-gnu
Thread model: posix
```

I'm seeing:

```
test-hexfloat.cc:109:7: warning: implicit conversion from 'unsigned int' to 'float' changes value from 4294967295 to 4294967296 [-Wimplicit-int-float-conversion]
      LOG_COMPLETION(bits);
      ^~~~~~~~~~~~~~~~~~~~
```

--- a/src/test-hexfloat.cc
+++ b/src/test-hexfloat.cc
@@ -35,15 +35,15 @@
   for (; bits >= last_bits;  \
        last_bits = bits, bits += num_threads_ * FOREACH_UINT32_MULTIPLIER)
 
-#define LOG_COMPLETION(bits)                                                 \
-  if (shard == 0) {                                                          \
-    int top_byte = bits >> 24;                                               \
-    if (top_byte != last_top_byte) {                                         \
-      printf("value: 0x%08x (%d%%)\r", bits,                                 \
-             static_cast<int>(static_cast<float>(bits) * 100 / UINT32_MAX)); \
-      fflush(stdout);                                                        \
-      last_top_byte = top_byte;                                              \
-    }                                                                        \
+#define LOG_COMPLETION(bits)                                                  \
+  if (shard == 0) {                                                           \
+    int top_byte = bits >> 24;                                                \
+    if (top_byte != last_top_byte) {                                          \
+      printf("value: 0x%08x (%d%%)\r", bits,                                  \
+             static_cast<int>(static_cast<double>(bits) * 100 / UINT32_MAX)); \
+      fflush(stdout);                                                         \
+      last_top_byte = top_byte;                                               \
+    }                                                                         \
   }
 
 #define LOG_DONE()     \

